infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -235,6 +226,23 @@ public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
         if (replicationContext.getIsLeader().get()) {
             removeStaleSessionOnLeadershipAcquire();
         }
+
+        // check if any new SINKs have been added to the topology.
+        Set<String> newRemoteSinksAdded = Sets.difference(newRemoteSinks, currentRemoteSinks);
+        if (!newRemoteSinksAdded.isEmpty()) {
+            configManager.getRegisteredSubscribers().forEach(this::createSessions);
+        } else {
+            // Since sessions are created on all the nodes in the SOURCE cluster, its possible that on leadership change,
+            // the new leader doesn't have any new sessions to create. In that case, only connect
+            connectToRemoteClusters();
+        }
+    }
+
+    private Set<LogReplicationSession> mergeIncomingOutgoingSessions() {
```

Comment:
can we rename this method to getAllSessions() or something similar?  It is performing a simple task of fetching all sessions but the name does not suggest that clearly.

Reply:
Sure, renaming it to "fetchAllSessions"